zetaclient/zetacore/client_monitor.go

```diff
@@ -165,6 +167,38 @@ func (c *Client) monitorVoteInboundResult(
 	return nil
 }
 
+func (c *Client) QueryBallot(ctx context.Context, msg *types.MsgVoteInbound) {
+	ballotIndex := msg.Digest()
+	ballot, err := c.GetBallot(ctx, ballotIndex)
+	if err != nil {
+		c.logger.Error().Err(err).Str("ballot.index", ballotIndex).Msg("QueryBallot: failed to query ballot")
+		return
+	}
+
+	totalVotes := len(ballot.Voters)
+	successVotes := 0
+	failVotes := 0
+	noVotes := 0
+	for _, vote := range ballot.Voters {
+		switch vote.VoteType {
+		case observertypes.VoteType_SuccessObservation:
+			successVotes++
+		case observertypes.VoteType_FailureObservation:
+			failVotes++
+		case observertypes.VoteType_NotYetVoted:
+			noVotes++
+		}
+	}
+
+	votes := map[string]any{
+		"total":   totalVotes,
+		"success": successVotes,
+		"fail":    failVotes,
+		"no_vote": noVotes,
+	}
+	c.logger.Debug().Fields(votes).Msgf("QueryBallot Before Executing with higher limit: ballot status %s", ballot.BallotStatus.String())
```

Comment:
you would need `Debug() -> Info()` to be able to print out the log message in the Datadog. Additionally, adding the `inboundHash` (and even together with `zetaTxHash`) to log fields would be very helpful.